proxy/httpproxy/reverse.go

```diff
@@ -110,15 +109,16 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 	var requestClosed int32
 	completeCh := make(chan bool, 1)
 	closeNotifier, ok := rw.(http.CloseNotifier)
-	cancel := httputil.RequestCanceler(proxyreq)
+	ctx, cancel := context.WithCancel(context.Background())
+	proxyreq = proxyreq.WithContext(ctx)
+	defer cancel()
 	if ok {
 		closeCh := closeNotifier.CloseNotify()
 		go func() {
 			select {
 			case <-closeCh:
 				atomic.StoreInt32(&requestClosed, 1)
 				plog.Printf("client %v closed request prematurely", clientreq.RemoteAddr)
-				cancel()
```

Comment:
we should still keep this cancel. the go routine exit at line 117 will not trigger the defer func at line 114.